google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java

```diff
@@ -0,0 +1,114 @@
+import com.google.ads.googleads.v3.services.CampaignDraftOperation;
+import com.google.ads.googleads.v3.services.CampaignDraftServiceClient;
+import com.google.ads.googleads.v3.services.MutateCampaignDraftResult;
+import com.google.ads.googleads.v3.services.MutateCampaignDraftsResponse;
+import com.google.ads.googleads.v3.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Adds a campaign draft for a campaign. Make sure you specify a campaign that has a non-shared
+ * budget.
+ */
+public class AddCampaignDraft {
+
+  private static class AddCampaignDraftParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.CAMPAIGN_ID, required = true)
+    private Long baseCampaignId;
+  }
+
+  public static void main(String[] args) {
+    AddCampaignDraftParams params = new AddCampaignDraftParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.baseCampaignId = Long.valueOf("INSERT_CAMPAIGN_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddCampaignDraft().runExample(googleAdsClient, params.customerId, params.baseCampaignId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId) {
+    // Creates an operation to create a new campaign draft.
+    CampaignDraftOperation.Builder opBuilder = CampaignDraftOperation.newBuilder();
```

Comment:
Why not just create the entire CampaignDraftOperation? Do we need a builder here?

Reply:
Builder is required since we call `getCreateBuilder()` so the result of chaining `build()` to the end would be `CampaignDraft` not `CampaignDraftOperation`.

That said, since this is hard to follow I've swapped it out.